frontend/src/components/common/form/Check.tsx

```diff
@@ -124,9 +124,9 @@ export const Check: React.FC<CheckProps> = ({
               placeholder={placeholder}
               checked={checked === false}
               onChange={(e: any) => {
-                setFieldTouched(field);
```

Comment:
so apparently setting touched like this is bad because it introduces a race condition between setfieldvalue and setFieldTouched. If setFieldTouched finishes first formik thinks the field has been touched but left empty which may result in validation errors.

Reply:
Yeah thats bad, best to set touched onBlur